includes/backends/class-omise-backend-installment.php

```diff
@@ -0,0 +1,127 @@
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_bbl' => array(
+				'bank_code'          => 'bbl',
+				'title'              => __( 'Bangkok Bank', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 500.00,
+			),
+
+			'installment_kbank' => array(
+				'bank_code'          => 'kbank',
+				'title'              => __( 'Kasikorn Bank', 'omise' ),
+				'interest_rate'      => 0.0065,
+				'min_allowed_amount' => 500.00,
+			),
+		);
+	}
+
+	/**
+	 * @param  string $currency
+	 * @param  float  $purchase_amount
+	 *
+	 * @return array  of an available installment providers
+	 */
+	public function get_available_providers( $currency, $purchase_amount ) {
+		// Note: as installment payment at the moment only supports for THB currency, so the 
+		//       $purchase_amount is multiplied with 100 to convert the amount into subunit (satang).
+		$providers = $this->capabilities()->getInstallmentBackends( $currency, ( $purchase_amount * 100 ) );
+
+		foreach ( $providers as &$provider ) {
+			$provider_detail = self::$providers[ $provider->_id ];
+
+			$provider->provider_code   = str_replace( 'installment_', '', $provider->_id );
+			$provider->provider_name   = isset( $provider_detail ) ? $provider_detail['title'] : strtoupper( $provider->code );
+			$provider->interest_rate   = $this->capabilities()->is_zero_interest() ? 0 : ( $provider_detail['interest_rate'] * 100 );
+			$provider->available_plans = $this->get_available_plans(
+				$purchase_amount,
+				$provider->allowed_installment_terms,
+				$provider->interest_rate,
+				$provider_detail['min_allowed_amount']
+			);
+		}
+
+		return $providers;
+	}
+
+	/**
+	 * @param  float $purchase_amount
+	 * @param  array $allowed_installment_terms
+	 * @param  float $interest_rate
+	 * @param  float $min_allowed_amount
+	 *
+	 * @return array  of an filtered available terms
+	 */
+	public function get_available_plans( $purchase_amount, $allowed_installment_terms, $interest_rate, $min_allowed_amount ) {
+		$plans = array();
+
+		sort( $allowed_installment_terms );
+
+		foreach ( $allowed_installment_terms as $term_length ) {
+			$monthly_amount = $this->calculate_monthly_payment_amount( $purchase_amount, $term_length, $interest_rate );
+
+			if ( $monthly_amount < $min_allowed_amount ) {
+				break;
+			}
+
+			$plans[] = array(
+				'term_length'    => $term_length,
+				'monthly_amount' => $monthly_amount
+			);
+		}
+
+		return $plans;
+	}
+
+	/**
+	 * @param  float $purchase_amount
+	 * @param  int   $term_length      A length of a given installment term.
+	 * @param  float $interest_rate    its value can be '0' if merchant absorbs an interest.
+	 *
+	 * @return float  of a installment monthly payment (round up to 2 decimals).
+	 */
+	public function calculate_monthly_payment_amount( $purchase_amount, $term_length, $interest_rate ) {
+		$interest = $purchase_amount * $term_length * $interest_rate;
```

Comment:
This strikes me as wrong. Interest should just be `amount * interest rate` as I understand instalment interest:

If I read that I was going to pay 15% interest on an instalment plan when the cost of the item was say B100, I would expect to pay B115 in total, but it seems here that the interest rate shown is used to calculate the amount of interest paid each and every month. So, if I selected 10 months at 15%, I'd end up paying in total 2.5 times the actual price of the item

Reply:
I probably used the same formula, but are we sure it is correct?

Reply:
@jonrandy according to a screenshot that Var have tested with a couple banks (installment provider).
At 3,000 THB, interest rate 0.8%, for 10 months. Net total is 3,240 THB (324 THB per month).
The interest of installment takes the term length into account as well.

So no, in case of 100 THB item with interest rate 15% for 10 months, you are not paying in total 115 THB but 250 THB. The longer you split the bill, the more you pay.

But as your comment:
> Why do we have interest rates as a decimal in one location in the code, and suddenly changed to a percentage here?

I think this can be improved a bit.
I'll update this code to
1. Interest rate as percentage
```php
'installment_bay' => array(
	'bank_code'          => 'bay',
	'title'              => __( 'Krungsri', 'omise' ),
	'interest_rate'      => 0.8, // instead of 0.008
	'min_allowed_amount' => 300.00,
),
```

2. Remove multiply 100 at
```php
$provider->interest_rate      = $this->capabilities()->is_zero_interest() ? 0 : ( $provider_detail['interest_rate'] * 100 );
```

3. Then, here will be 
```php
$interest = $purchase_amount * $interest_rate * $term_length / 100;
```

Reply:
Says cheese and I'll push the update 🧀